File: engage-4 (Angular en Spring)/engage/src/app/models/project/pending-request.ts
import {User} from '../user/user.model';
import {Project} from './project';

export class PendingRequest {
  id: number;
  sendByUser: User;
  project: Project;


  constructor(sendByUser: User, project: Project) {
    this.sendByUser = sendByUser;
    this.project = project;
  }
}

File: engage-4 (Angular and Spring)/engageserver/src/main/java/app/models/project/PendingRequest.java
package app.models.project;

import app.models.user.User;
import com.fasterxml.jackson.annotation.JsonBackReference;

import javax.persistence.*;

@Entity
public class PendingRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    @ManyToOne
    private User sendByUser;
    @ManyToOne
    private Project project;

    public PendingRequest() {

    }

    public PendingRequest(long id, User sendByUser) {
        this.id = id;
        this.sendByUser = sendByUser;
    }

    public long getId() {
        return id;
    }

    public User getSendByUser() {
        return sendByUser;
    }

    public Project getProject() {
        return project;
    }

    @Override
    public String toString() {
        return "PendingRequest{" +
                "id=" + id +
                ", sendByUser=" + sendByUser +
                ", project=" + project +
                '}';
    }
}
